specification/profiling.md

```diff
@@ -0,0 +1,169 @@
+
+### Call Stack Span Association
+
+When a call stack is sampled during the execution of a span scope,
+the profiler MUST be able to associate the call stack to the span.
+This association SHOULD happen as close to the sampling point as feasible,
+but MAY occur later in a processing pipeline.
+
+After the association has been made, the `TraceId` and `SpanId` fields of the 
+LogRecord MUST be populated (see below).
+
+### Call Stack Ingest
+
+Call stacks MUST be ingested as [OpenTelemetry Logs](https://github.com/open-telemetry/opentelemetry-specification/tree/main/specification/logs).
+The logs containing profiling data MUST be sent via OTLP/gRPC. 
+
+By default, the agent MUST send logs to the value in the `OTEL_EXPORTER_OTLP_ENDPOINT` environment
+variable. If this variable is not set, then the agent must default back to the gRPC/OTLP
+default (https://localhost:4317).
+
+The agent must allow the destination for profiling logs to be overridden with 
+the environment variable SPLUNK_PROFILER_LOGS_ENDPOINT.
+
+Agents SHOULD reuse persistent gRPC/OTLP connections from other components (traces, metrics).
+
+## ResourceLogs
+
+ResourceLogs is the upstream protobuf data type. It MUST be populated with the attributes
+from the OpenTelemetry resource.
+
+#### InstrumentationLibraryLogs
+
+Each `ResourceLogs` instance has an instance of `InstrumenationLibraryLogs`. 
+For each `InstrumentationLibraryLogs` instance:
+
+* `name` - MUST be set to `otel.profiling`
+* `version` - MUST be set to `0.1.0` 
+
+#### LogRecord Common Attributes
+
+Inside each `InstrumentationLibraryLogs` instance is a list of `LogRecord` instances. For each 
+`LogRecord` instance:
+
+* `com.splunk.sourcetype` MUST be set to the value `otel.profiling`
+* `source.event.period` MUST contain the sampling period for call stack sampler
+* `source.event.name` OPTIONALLY can contain the name of the event that triggered the sampling
+
+#### LogRecord Fields
+
+* [Name](https://github.com/open-telemetry/opentelemetry-specification/blob/main/specification/logs/data-model.md#field-name)
+  MUST be set to the constant value `otel.profiling`.
+* [Time](https://github.com/open-telemetry/opentelemetry-specification/blob/main/specification/logs/data-model.md#field-timestamp)
+  MUST be set to the time that the call stack was sampled.
+* [TraceId](https://github.com/open-telemetry/opentelemetry-specification/blob/main/specification/logs/data-model.md#field-traceid)
+  MUST be populated when a call stack has been sampled within a span scope.
+* [SpanId](https://github.com/open-telemetry/opentelemetry-specification/blob/main/specification/logs/data-model.md#field-spanid)
+  MUST be populated when a call stack has been sampled within a span scope.
+* [Body](https://github.com/open-telemetry/opentelemetry-specification/blob/main/specification/logs/data-model.md#field-body)
+  MUST be populated with the line-terminated call stack (see below)
+
+##### Call Stack Format
```

Comment:
Have you been considering defining it as structured JSON or Protobuf instead? I am just thinking that maybe it would be easier to implement in other languages and it could be easier to review the format.

Reply:
Yeah, I am considering that or possibly just making it implementation specific and adding a form of content-type attribute. For now, I just went with what Java has provided (and what the backend expects currently). I would like to treat this kind of change as a revision to the spec later as needed.